### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Angr1st**
In this PR I updated all dependent crates versions. I also adjusted the example code to work with the newest versions of the stm32f1xx-hal crate.

### CODE REVIEW

File: src/properties.rs [hunk 1]
```diff
@@ -107,29 +107,16 @@ where
...
-    /// # let interface = FakeInterface {};
-    /// #
-    /// let disp = DisplayProperties::new(
-    ///     interface,
-    ///     DisplaySize::Display128x64,
-    ///     DisplayRotation::Rotate0,
-    /// );
-    /// assert_eq!(disp.get_dimensions(), (128, 64));
+    /// # use ssd1309::test_helpers::StubInterface;
+    ///# let interface = StubInterface;
```
Comment (antonok-edm) [inline src/properties.rs:111]: nit: consistent usage of `/// #`

File: src/lib.rs [hunk 2]
```diff
@@ -34,46 +34,57 @@
...
-//! extern crate panic_abort;
+//! extern crate panic_semihosting;
 //! extern crate ssd1309;
-//! extern crate stm32f103xx_hal as blue_pill;
-//!
-//! use blue_pill::delay::Delay;
-//! use blue_pill::i2c::{DutyCycle, I2c, Mode};
+//! extern crate stm32f1xx_hal as blue_pill;
+//! 
+//!
```
Comment (antonok-edm) [inline src/lib.rs:48]: nit: double newline here

File: src/test_helpers.rs
Comment (antonok-edm) [inline src/test_helpers.rs:1]: I'm not a big fan of adding undocumented public API. Luckily there is a neat trick to make shared code available to tests without exposing it from the crate.

Could you remove `mod test_helpers` from `lib.rs` and add this to the doctest instead?
```
/// # #[path = "test_helpers.rs"]
/// # mod test_helpers;
/// # use test_helpers::StubInterface;
```

**Review by antonok-edm — COMMENTED** [review]
Thanks! A few minor things and this should be good to go.

File: src/properties.rs (hunk 1 above)
Comment (Angr1st) [reply src/properties.rs:113]: Fixed.

File: src/lib.rs (hunk 2 above)
Comment (Angr1st) [reply src/lib.rs]: Fixed.

File: src/test_helpers.rs
Comment (Angr1st) [reply src/test_helpers.rs:1]: Done. Thanks for the suggestion 👍

**Review by antonok-edm — APPROVED** [review]
sorry for delay and thanks again!

**Pull request merged**